### DISCUSSION TITLE
Sign Up Email Confirmations

### DISCUSSION THREAD

**Discussion by notscott25** [supabase/supabase#7330] [Questions]
I have email confirmations turned on.
Given the docs that should mean:
If "Email Confirmations" is turned on, both a user and a session will be null

But a check of response.user.id provides a value?

if (response.error != null) {
        showMessage('Sign up failed: ${response.error!.message}');
        print(response.error!.message);
        _btnController.reset();
      } else if (response.data == null && response.user == null) {
        showMessage(
            "Please check your email and follow the instructions to verify your email address.");
        _btnController.success();
      }

As you can see here this would not have an error and neither the response.data nor the response.user are null

**Comment by GaryAustin1** [comment]
https://supabase.com/docs/reference/javascript/auth-signup#notes
"If "Email Confirmations" is turned on, a user is returned but session will be null"
Seems to say user is returned, but not session...

**Reply by notscott25** [reply]
It would seem the JavaScript docs and dart docs are different. Now I am wondering which is correct.

**Reply by GaryAustin1** [reply]
Just to wrap up in a complete thread.  
Dart documents are different than JS documents.
JS behavior matches JS docs, I verified it.
So if Dart behavior matches JS behavior then highly likely Dart docs are wrong and an issue should be generated.

**Comment by GaryAustin1** [comment]
Interesting... 
Dart signup doc says both are null...

**Reply by notscott25** [reply]
I hadn't reviewed the JavaScript docs.
So the JavaScript docs seem correct?
Or are the dart docs correct and this is a bug?

**Comment by GaryAustin1** [comment]
JS doc is correct for JS, I checked it.
If you are using Dart and it matches the JS behavior then the Dart docs are probably wrong.  I have no way to check it.
If you want to generate an issue that would be good, otherwise I will later.

**Reply by notscott25** [reply]
I can generate the issue that's fine. Thanks.
